### PR TITLE
corrected use of PSModulePath casing to be consistent with Windows PowerShell

### PR DESCRIPTION
**Pull request by SteveL-MSFT**
addresses #3227

### CODE REVIEW

File: src/System.Management.Automation/engine/Modules/ModuleIntrinsics.cs [hunk 1]
```diff
@@ -734,8 +734,8 @@ private static bool NeedToClearProcessModulePath(string currentProcessModulePath
                 // so if the current process module path contains any of them, it's likely that the sxs
                 // ps was started directly on windows, or from full ps. The same goes for the legacy personal
                 // and shared module paths.
-                string hklmModulePath = GetExpandedEnvironmentVariable("PSMODULEPATH", EnvironmentVariableTarget.Machine);
-                string hkcuModulePath = GetExpandedEnvironmentVariable("PSMODULEPATH", EnvironmentVariableTarget.User);
+                string hklmModulePath = GetExpandedEnvironmentVariable("PSModulePath", EnvironmentVariableTarget.Machine);
```
Comment (lzybkr) [inline src/System.Management.Automation/engine/Modules/ModuleIntrinsics.cs]: I count 8 references to this same string in code - there should be a `const string` that we use instead.

File: test/powershell/engine/Remoting/SessionOption.Tests.ps1 [hunk 2]
```diff
@@ -1,3 +1,4 @@
+return
```
Comment (iSazonov) [inline test/powershell/engine/Remoting/SessionOption.Tests.ps1]: It seems we should remove.

File: test/powershell/Language/Classes/Scripting.Classes.Modules.Tests.ps1 [hunk 3]
```diff
@@ -76,9 +76,9 @@ Describe 'Module reloading with Class definition' -Tags "CI" {
 
     BeforeAll {
         Set-Content -Path TestDrive:\TestModule.psm1 -Value @'
-$passedArgs = $args
-class Root { $passedIn = $passedArgs }
-function Get-PassedArgsRoot { [Root]::new().passedIn }
+$passedArgs = $args
+class Root { $passedIn = $passedArgs }
+function Get-PassedArgsRoot { [Root]::new().passedIn }
```
Comment (iSazonov) [inline test/powershell/Language/Classes/Scripting.Classes.Modules.Tests.ps1]: What's changed?

File: src/System.Management.Automation/engine/Modules/ModuleIntrinsics.cs (hunk 1 above)
Comment (SteveL-MSFT) [inline src/System.Management.Automation/engine/Modules/ModuleIntrinsics.cs]: ok

File: test/powershell/engine/Remoting/SessionOption.Tests.ps1 (hunk 2 above)
Comment (SteveL-MSFT) [inline test/powershell/engine/Remoting/SessionOption.Tests.ps1]: that was accidentally committed as those tests were failing due to external issue, will remove

File: test/powershell/Language/Classes/Scripting.Classes.Modules.Tests.ps1 (hunk 3 above)
Comment (SteveL-MSFT) [inline test/powershell/Language/Classes/Scripting.Classes.Modules.Tests.ps1]: Didn't change anything here as PSModulePath isn't here.  Not sure why this is showing up in the PR as a change.

File: src/System.Management.Automation/engine/PropertyAccessor.cs [hunk 4]
```diff
@@ -518,11 +518,11 @@ internal override string GetModulePath(PropertyScope scope)
         {
             if (PropertyScope.CurrentUser == scope)
             {
-                return ModuleIntrinsics.GetExpandedEnvironmentVariable("PSMODULEPATH", EnvironmentVariableTarget.User);
+                return ModuleIntrinsics.GetExpandedEnvironmentVariable("PSModulePath", EnvironmentVariableTarget.User);
```
Comment (lzybkr) [inline src/System.Management.Automation/engine/PropertyAccessor.cs]: `Constants.PSModulePathEnvVar` - right?

Comment (lzybkr) [inline src/System.Management.Automation/engine/PropertyAccessor.cs]: Whitespace changes - add `?w=1` to see the url to see diffs ignoring whitespace.

Comment (SteveL-MSFT) [inline src/System.Management.Automation/engine/PropertyAccessor.cs]: Did a search for `"PSModulePath"` and I think I got them all

File: test/powershell/Language/Classes/Scripting.Classes.Modules.Tests.ps1 (hunk 3 above)
Comment (SteveL-MSFT) [inline test/powershell/Language/Classes/Scripting.Classes.Modules.Tests.ps1]: Don't see any whitespace difference either

**Review by TravisEz13 — APPROVED** [review]

**Comment by SteveL-MSFT** [comment]
@TravisEz13 merge?

**Pull request merged**